xc7/utils/prjxray_routing_import.py

```diff
@@ -223,6 +231,35 @@ def check_feature(feature):
     .ENABLE_BUFFER.HCLK_CK_BUFHCLK10
     """
 
+    # IOI_SING tiles have bits in common with the IOI tiles.
+    #
+    # The difference is that the TOP IOI_SING tile shares bits with
+    # the bottom half of a normal IOI tile, while the BOTTOM IOI_SING
+    # shares bits with the top half of a normal IOI TILE.
+    #
+    # The following, is to change the edge feature to accomodate this
+    # need, as the IOI_SING tiles have the same wire, and pip names
+    # despite they are found on the TOP or BOTTOM of an IOI column
+    m = IOI_SING.fullmatch(feature)
+    if m:
+        is_bottom_sing = int(m.group(2)) % 50 == 0
```

Comment:
I'd add a comment why 50 since it is a hard-coded constant.

Reply:
Done